### PR TITLE
BUILD: Harmonize deps across multiple env files

### PR DESCRIPTION
**Pull request by MridulS**

### CODE REVIEW

**Comment by MridulS** [comment]
The build is failing as plopp 24.09.0 isn't up on conda yet.

File: docs/environments/developer.yml [hunk 1]
```diff
@@ -25,13 +25,13 @@ dependencies:
   - pooch
   - pre-commit
   - pytest
-  - python=3.10
+  - python=3.12
```
Comment (jl-wynen) [inline docs/environments/developer.yml:28]: Why? We should develop with the minimal supported version to avoid accidentally using newer features.

Comment (MridulS) [reply docs/environments/developer.yml:28]: Hmm, I would personally prefer the CI catch these things but I won't fight it :)

Comment (jl-wynen) [reply docs/environments/developer.yml:28]: That is later than catching it locally. So it's more work. Plus, when you lock dependencies, you have to use the same Python version locally.

Comment (MridulS) [reply docs/environments/developer.yml:28]: ah yeah true!

**Review by jl-wynen — APPROVED** [review]

**Comment by MridulS** [comment]
I had to revert the bump to gtest, windows build was failing. I did try some debugging but for now I have reverted the bump.

But atleast now this build with windows-2022 image. We need to update the requried actions text to merge this in.

File: lib/cmake/scipp-conan.cmake [hunk 2]
```diff
@@ -16,10 +16,10 @@ if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/conan.cmake")
   )
   file(
     DOWNLOAD
-    "https://raw.githubusercontent.com/conan-io/cmake-conan/0.17.0/conan.cmake"
+    "https://raw.githubusercontent.com/conan-io/cmake-conan/6e5369d13720f22e07e31bb6b9018dbe60529fea/conan.cmake"
```
Comment (jl-wynen) [inline lib/cmake/scipp-conan.cmake:19]: What version is this?

Comment (MridulS) [reply lib/cmake/scipp-conan.cmake:19]: The current HEAD of the main branch for conan1.X, there hasn't been a release in some time.

**Review by jl-wynen — APPROVED** [review]

**Comment by MridulS** [comment]
We need to fix the text in the required tests to use `windows-2022` instead of `windows-2019` to merge this PR.

**Comment by jl-wynen** [comment]
Done

**Pull request merged**